src/Interpreter/ExprEval.cpp

```diff
@@ -327,27 +327,33 @@ LLVMValue ExprEvaluator::visitGlobalVariable(llvm::GlobalVariable& global) {
     throw Unevaluatable(&global, "global had no initializer");
   }
 
-  OpRef data =
-      visitGlobalData(*global.getInitializer(), global.getAddressSpace());
-  const auto& array = llvm::cast<ArrayBase>(*data);
-
   const llvm::DataLayout& layout = ctx->mod->getDataLayout();
   unsigned bitwidth = layout.getPointerSizeInBits();
-  unsigned alignment = global.getAlignment();
+  llvm::Type* type = global.getInitializer()->getType();
+  OpRef size = ConstantInt::Create(
+      llvm::APInt(bitwidth, layout.getTypeAllocSize(type).getFixedSize()));
+  OpRef alloc_data = AllocOp::Create(size, ConstantInt::CreateZero(8));
+
   auto perms = global.isConstant() ? AllocationPermissions::Write
                                    : AllocationPermissions::ReadWrite;
 
+  unsigned alignment = global.getAlignment();
+
   auto alloc = ctx->heaps[global.getAddressSpace()].allocate(
-      array.size(), ConstantInt::Create(llvm::APInt(bitwidth, alignment)), data,
+      size, ConstantInt::Create(llvm::APInt(bitwidth, alignment)), alloc_data,
       AllocationKind::Global, perms, *ctx);
 
-  auto pointer = LLVMValue(
+  auto ptr =
       Pointer(alloc, ConstantInt::Create(llvm::APInt::getNullValue(bitwidth)),
```

Comment:
I realize you haven't changed this but using `CreateZero` would be better here.
```suggestion
      Pointer(alloc, ConstantInt::CreateZero(bitwidth),
```